src/components/Nav/index.tsx

```diff
@@ -92,16 +104,44 @@ const Nav: FC<IProps> = ({ path }) => {
                   variant="ghost"
                   isSecondary
                   px={1.5}
+                  _hover={{
+                    transform: "rotate(30deg)", // Rotate the icon to 30 degrees on hover
+                    color: "blue", // Change color to blue on hover
+                  }}
                   onClick={toggleColorMode}
                 ></IconButton>
+
                 <ButtonLink
+                  onMouseOver={() => setLanguagesHover(true)}
+                  onMouseOut={() => setLanguagesHover(false)}
+                  as={Link}
                   to={`/languages/${fromPageParameter}`}
-                  leftIcon={<Icon as={MdLanguage} />}
                   variant="ghost"
                   isSecondary
                   px={1.5}
+                  _hover={{
+                    color: "blue", // Change color to blue on hover
+                  }}
+                  style={{
+                    display: "inline-flex",
+                    alignItems: "center",
+                  }}
                 >
-                  {t("languages")} {i18n.language.toUpperCase()}
+                  <Box mr={2} mt={2}>
+                    {" "}
+                    {/* Add spacing between the text and the icon */}
+                    <Icon
+                      as={MdLanguage}
+                      style={{
+                        transition: "transform 0.3s ease-in-out, color 0.2s", // Apply the transition to the icon
+                        transform: languagesHover
+                          ? "rotate(30deg)"
+                          : "rotate(0deg)", // Rotate the icon to 30 degrees on hover
+                        color: languagesHover ? "blue" : "inherit", // Change color to blue on hover
```

Comment:
This needs to be removed, and will be inherited by the hover state supplied directly to `ButtonLink`
```suggestion
```